### PR TITLE
cluster-ui: update nodes label for single regions

### PR DESCRIPTION
**Pull request by xinhaoz**
For single region clusters instead of grouping
nodes by region we'll just display the list of
nodes, with a max of 4 being displayed in a cell
at a time. Overflow will be shown in a tooltip
on hover.

Epic: CRDB-37558
Release note: None

----------------------------

ui: fix store ids to node regions mapping

This commit fixes the store ids to node regions
mapping for the new db pages. Previously, we
mapped each store id to its distinct entry
in the nodesByRegionsMap which did not consider
that the set of store ids can be bigger than the
set of node ids. This lead to duplicate node entries
in the map.

Epic: CRDB-37558
Release note: None

### CODE REVIEW

**Comment by xinhaoz** [comment]
image
Pasted Graphic 1

File: pkg/ui/workspaces/cluster-ui/src/components/regionNodesLabel/components/nodesList.tsx [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+import { NodeID } from "src/types/clusterTypes";
+
+import styles from "./nodesList.module.scss";
+
+type Props = {
+  nodes: NodeID[];
+};
+
+export const NodesList: React.FC<Props> = ({ nodes }) => {
+  const displayedNodes = nodes.slice(0, 4);
```
Comment (kyle-a-wong) [inline pkg/ui/workspaces/cluster-ui/src/components/regionNodesLabel/components/nodesList.tsx:19]: The row below has `nodes?.length`. If nodes can be undefined, does this also need to be: `nodes?.slice(0, 4) ?? []`

Comment (xinhaoz) [reply pkg/ui/workspaces/cluster-ui/src/components/regionNodesLabel/components/nodesList.tsx:19]: Added a default value so the optional check isn't necessary.

**Review by kyle-a-wong — APPROVED** [review]
:lgtm:

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/132312#-:-O8w6dK994VXhXUY64V3:bnfp4nl)* status: <span>:shipit:</span> complete! 1 of 0 LGTMs obtained (waiting on @angles-n-daemons and @dhartunian)

**Comment by xinhaoz** [comment]
TFTR!
bors r+

**Pull request merged**